ntcore/src/main/native/cpp/ntcore_test.cpp

```diff
@@ -147,10 +148,8 @@ struct NT_Value* NT_GetValueStringArrayForTesting(uint64_t last_change,
   value->data.arr_string.size = array_len;
   for (size_t i = 0; i < value->data.arr_string.size; ++i) {
     size_t len = arr[i].len;
-    value->data.arr_string.arr[i].len = len;
-    value->data.arr_string.arr[i].str =
-        static_cast<char*>(wpi::safe_malloc(len + 1));
-    std::memcpy(value->data.arr_string.arr[i].str, arr[i].str, len + 1);
+    auto write = WPI_AllocateString(&value->data.arr_string.arr[i], len);
+    std::memcpy(write, arr[i].str, len + 1);
```

Comment:
I realized [my previous comment](https://github.com/wpilibsuite/allwpilib/pull/6299/files#r1465688690) might not have been clear- the memcpy here and in [`ntcore_c.cpp#L927`](https://github.com/ThadHouse/allwpilib/blob/90e915d5545fab98553cbcc1209897bc80806b7d/ntcore/src/main/native/cpp/ntcore_c.cpp#L927) does use `len + 1`, and I was wondering why it did so.

Reply:
Ah. Yeah the github comment hid that, so I need to fix those.